src/app/shared/enums/checkpoint-type.ts

```diff
@@ -0,0 +1,23 @@
+export const enum CheckpointType {
+  Startpoint = 1,
+  Endpoint,
+  Abortpoint,
+  Inputpoint,
+  Outputpoint,
+  Infopoint,
+  ThreadStartpointError,
+  ThreadStartpoint,
+  ThreadEndpoint,
+}
```

Comment:
We shouldn't use enums: https://github.com/frankframework/frontend-conventions/blob/master/typescript.md#enums

Reply:
I know, and I discussed this with @Matthbo but neither of us could figure out a good way that follows our conventions. Have a look at the way it's implemented and tell me how we should do it.

Reply:
You could change the enum to use  `as const` instead but then you manually have to set the value for all the consecutive values. We discussed using an array instead so we could use the index as value but @MatthijsSmets didnt like the the `0` index being null/undefined.

Reply:
> but @MatthijsSmets didnt like the the 0 index being null/undefined.

Me neither. 

> but then you manually have to set the value for all the consecutive values.

That's unfortunate. I think it's a bit weird, tho, to have an exception to the rule. With all the type safety and such.

Reply:
I'm not a fan either, but the old solution was way worse.

Reply:
I sent an hour last friday getting something better to work and it either got way too complex for this issue or I couldn't find a way to get it working properly with ts typings

Reply:
+1